src/common/io/src/geometry.rs

```diff
@@ -119,3 +168,60 @@ pub fn parse_to_subtype(buf: &[u8]) -> Result<GeometryDataType> {
         }
     }
 }
+
+pub fn read_ewkb_srid<R: Read>(raw: &mut R) -> std::result::Result<Option<i32>, io::Error> {
+    let byte_order = raw.ioread::<u8>()?;
+    let is_little_endian = byte_order != 0;
+    let endian = Endian::from(is_little_endian);
+    let type_id = raw.ioread_with::<u32>(endian)?;
+    let srid = if type_id & GEO_TYPE_ID_MASK == GEO_TYPE_ID_MASK {
+        Some(raw.ioread_with::<i32>(endian)?)
+    } else {
+        None
+    };
+
+    Ok(srid)
+}
+
+pub trait GeometryFormatOutput {
+    fn format(self, data_type: GeometryDataType) -> Result<String>;
+}
+impl<B: AsRef<[u8]>> GeometryFormatOutput for Ewkb<B> {
+    fn format(self, format_type: GeometryDataType) -> Result<String> {
+        match format_type {
+            GeometryDataType::WKB => self
+                .to_wkb(CoordDimensions::xy())
+                .map(|bytes| {
+                    bytes
+                        .iter()
+                        .map(|b| format!("{:02X}", b))
```

Comment:
better to use `hex::encode_upper` function to convert binary to hex format for better display.